### ISSUE TITLE
Following Getting Started Crashes with Simple Apollo Server

### ISSUE THREAD

**Issue by mikecann** [dotansimha/graphql-code-generator#1481]
Hi,

I have a very basic Apollo server (literally hello world using "apollo-server-express": "^2.4.8"), followed the very first steps of the guide (using gql-gen init) from here: https://graphql-code-generator.com/docs/getting-started/

Upon running `gql-gen --config codegen.yml` I get the following error:

```
$ yarn generate
yarn run v1.12.3
$ gql-gen --config codegen.yml
  √ Parse configuration
  > Generate outputs
    > Generate src/generated/graphql.ts
      × Load GraphQL schemas
        → write EPROTO 14220:error:1408F10B:SSL routines:ssl3_get_record:wrong version numb
e…
        Load GraphQL documents
        Generate
    > Generate ./graphql.schema.json
      × Load GraphQL schemas
        → write EPROTO 14220:error:1408F10B:SSL routines:ssl3_get_record:wrong version numb
e…
        Load GraphQL documents
        Generate


  Found 2 errors

   × src/generated/graphql.ts
    Error: write EPROTO 14220:error:1408F10B:SSL routines:ssl3_get_record:wrong version num
ber:openssl\ssl\record\ssl3_record.c:252:

   × ./graphql.schema.json
    Error: write EPROTO 14220:error:1408F10B:SSL routines:ssl3_get_record:wrong version num
ber:openssl\ssl\record\ssl3_record.c:252:


Something went wrong
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Package.json:
```
{
  "name": "my-apollo-server",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "dev": "nodemon -w src/**/*.ts --exec ts-node --files ./src/index.ts",
    "generate": "gql-gen --config codegen.yml"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "@types/node": "^11.11.3",
    "apollo-server-express": "^2.4.8",
    "express": "^4.16.4",
    "express-session": "^1.15.6",
    "graphql": "^14.1.1",
    "graphql-codegen-typescript-server": "^0.18.0",
    "nodemon": "^1.18.10",
    "ts-node": "^8.0.3",
    "typescript": "^3.3.3333"
  },
  "devDependencies": {
    "graphql-code-generator": "^0.18.0",
    "graphql-codegen-introspection": "^0.18.0",
    "graphql-codegen-typescript-common": "^0.18.0",
    "husky": "^1.3.1",
    "prettier": "^1.16.4",
    "pretty-quick": "^1.10.0"
  },
  "husky": {
    "hooks": {
      "pre-commit": "pretty-quick --staged"
    }
  }
}
```

codegen.yml
```
overwrite: true
schema: "https://localhost:4000"
documents: null
generates:
  src/generated/graphql.ts:
    plugins:
      - "typescript-common"
  ./graphql.schema.json:
    plugins:
      - "introspection"
```

index.ts
```
import express from "express";
import { ApolloServer, gql } from "apollo-server-express";

const typeDefs = gql`
  type Query {
    hello: String
  }
`;

// Provide resolver functions for your schema fields
const resolvers = {
  Query: {
    hello: () => "Hello world!"
  }
};

const server = new ApolloServer({ typeDefs, resolvers });

const app = express();
server.applyMiddleware({ app });

app.listen({ port: 4000 }, () =>
  console.log(`🚀 Server ready at http://localhost:4000${server.graphqlPath}`)
);
```

**Comment by mikecann** [comment]
Ah.. Okay I think I have worked it out. Its an SSL error so the config schema should read this:

`schema: "http://localhost:4000/graphql"` perhaps the cli `init` should hint at this?

the cli also defaults to the generated files going in `src/generated/graphql.ts` but then errors if that folder doesnt exist, it probably should create the folder for you no?

**Comment by dotansimha** [comment]
@mikecann I create an issue for the directory: https://github.com/dotansimha/graphql-code-generator/issues/1482 

Regarding the SSL error, how can we handle it? It looks like your GraphQL endpoint should work with SSL, but something it wrong with it's configuration.

**Comment by mikecann** [comment]
@dotansimha not sure. I guess this error just looks odd: 

```
Error: write EPROTO 14220:error:1408F10B:SSL routines:ssl3_get_record:wrong version num
ber:openssl\ssl\record\ssl3_record.c:252:
```

It probably should be something like "cannot find endpoint, are you sure you have the path to the graphql server correct" or something instead?

But not a big deal. Feel free to close.

**Comment by ENYOvest** [comment]
This might help someone. 
https://www.spektor.dev/apollo-client-and-ssl-handshake-failure/
Credits to Yossi Spektor.

TLDR. passing the entire header to apollo is the exact cause of this error in my case
